src/test/java/jewellery/inventory/unit/service/security/AuthServiceTest.java

```diff
@@ -0,0 +1,112 @@
+  @Mock private UserRepository userRepository;
+  @Mock private AuthenticationManager authenticationManager;
+  @Mock private JwtTokenService jwtTokenService;
+  @Mock private UserMapper userMapper;
+
+  private AuthenticationRequestDto authRequest;
+  private User user;
+
+  @BeforeEach
+  public void setUp() {
+    authRequest = new AuthenticationRequestDto();
+    authRequest.setEmail(USER_EMAIL);
+    authRequest.setPassword(USER_PASSWORD);
+
+    user = new User();
+    user.setId(UUID.randomUUID());
+    user.setEmail(USER_EMAIL);
+    user.setPassword(USER_PASSWORD);
+  }
+
+  @Test
+  void authenticateWithValidCredentialsReturnsToken() {
+    when(authenticationManager.authenticate(any())).thenReturn(null);
+    when(userRepository.findByEmail(authRequest.getEmail())).thenReturn(Optional.of(user));
+    when(jwtTokenService.generateToken(user)).thenReturn(TEST_TOKEN);
+
+    UserAuthDetailsDto response = authService.login(authRequest);
+
+    assertEquals(TEST_TOKEN, response.getToken());
+  }
+
+  @Test
+  void authenticateWithInvalidCredentialsThrowsInvalidCredentialsException() {
+    user.setPassword("wrongPassword");
+    when(authenticationManager.authenticate(any())).thenThrow(new InvalidCredentialsException());
+    when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(user));
+
+    assertThrows(InvalidCredentialsException.class, () -> authService.login(authRequest));
+  }
+
+  @Test
+  void authenticateWithNonExistentUserThrowsUserNotFoundException() {
+    when(userRepository.findByEmail(authRequest.getEmail())).thenReturn(Optional.empty());
+
+    assertThrows(UserNotFoundException.class, () -> authService.login(authRequest));
+  }
+
+  @Test
+  void verifyMethodsCalledWithExpectedParameters() {
+    when(authenticationManager.authenticate(any())).thenReturn(null);
+    when(userRepository.findByEmail(USER_EMAIL)).thenReturn(Optional.of(user));
+    when(jwtTokenService.generateToken(user)).thenReturn(TEST_TOKEN);
+
+    authService.login(authRequest);
+
+    verify(authenticationManager)
+        .authenticate(new UsernamePasswordAuthenticationToken(USER_EMAIL, USER_PASSWORD));
+    verify(userRepository).findByEmail(USER_EMAIL);
+    verify(jwtTokenService).generateToken(user);
+    verify(userMapper)
+        .toUserResponse(user);
+  }
+
+  @Test
+  void shouldReturnExpectedUserResponse() {
+    when(authenticationManager.authenticate(any())).thenReturn(null);
+    when(userRepository.findByEmail(authRequest.getEmail())).thenReturn(Optional.of(user));
+    when(jwtTokenService.generateToken(user)).thenReturn(TEST_TOKEN);
+
+    UserResponseDto userResponseDto = new UserResponseDto();
+    userResponseDto.setEmail(USER_EMAIL);
+
+    when(userMapper.toUserResponse(user)).thenReturn(userResponseDto);
+
+    UserAuthDetailsDto response = authService.login(authRequest);
+
+    assertEquals(userResponseDto, response.getUser());
+    assertEquals(TEST_TOKEN, response.getToken());
+  }
```

Comment:
Same question with this method. Is the setup any different than `authenticateWithValidCredentialsReturnsToken`? What are we expecting differently?

Reply:
The first test was written before the refactor of the response we are returning. At first we returned just the token so we tested just for that. The second test was written after the refactor and in this case we check if the user data returned is correct as well. But you are correct it should be only one test. Fixed by merging.